include/h2o/memory.h

```diff
@@ -182,11 +182,14 @@ typedef struct st_h2o_doublebuffer_t {
 extern void *(*volatile h2o_mem__set_secure)(void *, int, size_t);
 
 /**
- * prints an error message and aborts
+ * prints an error message and aborts. h2o_fatal can be modified by
+ * setting the function pointer it expands to, which is
+ * h2o__fatal_redirect.
  */
 H2O_NORETURN void h2o__fatal(const char *file, int line, const char *msg, ...) __attribute__((format(printf, 3, 4)));
+extern H2O_NORETURN void (*h2o__fatal_redirect)(const char *file, int line, const char *msg, ...);
```

Comment:
Thank you for the pull request. I think this is good change, but could we keep the name same? I.e.,

```suggestion
extern H2O_NORETURN void (*h2o__fatal)(const char *file, int line, const char *msg, ...);
```